tests/bignum-test.test.ts

```diff
@@ -0,0 +1,57 @@
+import { assertPrint, assertTCFail, assertTC, assertFail } from "./asserts.test";
+import { NUM, BOOL, NONE, CLASS } from "./helpers.test"
+
+export function test_print(n: bigint) {
+    assertPrint("1_test_print", 
+        `print(${n})`, 
+        [`${n}`]);
+}
+
+export function test_init_assign(x_init: bigint, y_init: bigint, x_val: bigint) {
+    assertPrint("2_test_init_assign", 
+        `x : int = ${x_init}
+        y : int = ${y_init}
+        y = x
+        print(y)
+        x = ${x_val}
+        print(x)
+        print(y)`,
+        [`${x_init}`, `${x_val}`, `${x_init}`]);
+}
+
+export function test_binop(des: string, x: bigint, y: bigint, op: string, res: string) {
+    assertPrint(des, 
+        `x : int = ${x}
+        y : int = ${y}
+        print(y ${op} x)`, 
+        [`${res}`]);
+}
+
+export function test_uniop(des: string, x: bigint, op: string, res: string) {
+    assertPrint(des, 
+        `x : int = ${x}
+        print(${op}x)`, 
+        [`${res}`]);
+}
+
+export function test_param_return(des: string, c: bigint, param: bigint, res: string) {
+    assertPrint(des, 
+        `def f(c:int) -> int:
+            c = ${c}
+            return c

```

Comment:
Thank you for making tests in functions. It would be useful to make sure the _other_ operations that were tested at the top level also work in functions. Since you've done the work of the helpers here (which is nice), maybe there's a way to test this even more by plugging the same tests into different contexts?

Reply:
We added a couple of test cases including the ones you mentioned in our design proposal. All binary operations are tested wrapped in function declarations and works with negative values.